ext/hal/nordic/CMakeLists.txt

```diff
@@ -11,16 +11,23 @@ if(CONFIG_HAS_NRFX)
   zephyr_include_directories(nrfx/mdk)
   zephyr_include_directories(.)
 
+  # Connect Kconfig compilation option for Non-Secure software with option required by MDK/nrfx
+  if (CONFIG_ARM_NONSECURE_FIRMWARE)
+     zephyr_compile_definitions(NRF_TRUSTZONE_NONSECURE)
+  endif()
```

Comment:
This should be moved further down, and use the same API as the below defines for consistency.

zephyr_compile_definitions_ifdef(CONFIG_ARM_NONSECURE_FIRMWARE NRF_TRUSTZONE_NONSECURE)

Reply:
Thanks, @SebastianBoe , I am going to rework this